benchmark_test.go

```diff
@@ -0,0 +1,101 @@
+package humanlog
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/humanlogio/humanlog/pkg/sink/bufsink"
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkHarness(b *testing.B) {
+	ctx := context.Background()
+	root := "test/benchmark"
+	des, err := os.ReadDir(root)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	for _, de := range des {
+		if !de.IsDir() {
+			continue
+		}
+
+		dir := filepath.Join(root, de.Name())
+		fileName, err := findfirstMatchedFileName(dir, "*.gz")
+		require.NoError(b, err)
+
+		b.ResetTimer()
+		testCase := dir
+		b.Run(testCase, func(bb *testing.B) {
+			path := filepath.Join(dir, fileName)
+			f, err := os.Open(path)
+			require.NoError(bb, err)
+			defer f.Close()
+
+			gzipReader, err := gzip.NewReader(f)
+			require.NoError(bb, err)
+
+			buf := make([]byte, 0, 100*1024) // set initial capacity to 100kb
+			err = loadAll(&buf, gzipReader)
+
+			src := bytes.NewBuffer(buf)
+			require.NoError(bb, err)
+
+			sink := bufsink.NewSizedBufferedSink(len(buf), nil) // size must be greater than src size
```

Comment:
len(buf) is probably way too big

you don't need to buffer the result, you can create a new "NopSink" that discards the log events entirely so that the benchmark can focus on the scanner.

Reply:
Will `NopSink` be used one day in the future? For now, I just defined `NopSink` rather than making a new pacakge.

Reply:
we can move it into another package if we need it later, but for now its good to keep it here only